src/code_f2470_len_27f0.c

```diff
@@ -191,26 +191,150 @@ INCLUDE_API_ASM(code_f2470_len_27f0, SetNpcFlagBits);
 
 INCLUDE_API_ASM(code_f2470_len_27f0, GetNpcPos);
 
-INCLUDE_API_ASM(code_f2470_len_27f0, EnableNpcShadow);
+ApiStatus EnableNpcShadow(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* ptrReadPos = script->ptrReadPos;
+    NpcId npcID = get_variable(script, *ptrReadPos++);
+    s32 enableShadow = get_variable(script, *ptrReadPos++);
+    Npc* npcPtr = resolve_npc(script, npcID);
 
-INCLUDE_API_ASM(code_f2470_len_27f0, EnableNpcBlur);
+    if (npcPtr != NULL) {
+        if (enableShadow) {
+            enable_npc_shadow(npcPtr);
+            return ApiStatus_DONE2;
+        }
+        else {
+            disable_npc_shadow(npcPtr);
+        }
+    }
+    return ApiStatus_DONE2;
+}
+
+ApiStatus EnableNpcBlur(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* ptrReadPos = script->ptrReadPos;
+    NpcId npcID = get_variable(script, *ptrReadPos++);
+    s32 enableBlur = get_variable(script, *ptrReadPos++);
+    Npc* npcPtr = resolve_npc(script, npcID);
+
+    if (npcPtr != NULL) {
+        if (enableBlur) {
+            enable_npc_blur(npcPtr);
+            return ApiStatus_DONE2;
+        }
+        else {
```

Comment:
The control flow of these are weird...but if we need the early return, we don't need an else probably

Reply:
Yeah, the returns are required. I removed the `else`.